src/openttd.cpp

```diff
@@ -875,6 +876,7 @@ static void MakeNewGameDone()
 		c->colour = _settings_client.gui.starting_colour;
 		ResetCompanyLivery(c);
 		_company_colours[c->index] = c->colour;
+		BuildOwnerLegend();
```

Comment:
I think this should use `CMD_SET_COMPANY_COLOUR`, just like for the second CC below.